packages/next-plugin/src/index.ts

```diff
@@ -115,15 +145,9 @@ export const createVanillaExtractPlugin = (
         dir,
         resolvedNextConfig.experimental?.appDir ?? false,
       );
-      const hasAppDir =
-        !!resolvedNextConfig.experimental?.appDir &&
-        !!(findPagesDirResult && findPagesDirResult.appDir);
+      const hasAppDir = !!(findPagesDirResult && findPagesDirResult.appDir);
 
-      const outputCss = hasAppDir
-        ? // Always output css since Next.js App Router needs to collect Server CSS from React Server Components
-          true
-        : // There is no appDir, do not output css on server build
-          !isServer;
+      const outputCss = true;
```

Comment:
for reference, if `outputCss` is false on the server (using pages router), the class created by the globalStyle below won't be applied to the ssr'd html

```ts
export const styleCompositionInSelector = style([
  style({ color: 'white' }),
  style({ backgroundColor: 'black' }),
]);

globalStyle(`body ${styleCompositionInSelector}`, {
  fontSize: '24px',
});
```

it *will* work if the selector looks like this

```ts
export const styleCompositionInSelector = style([
  {},
  style({ color: 'white' }),
  style({ backgroundColor: 'black' }),
]);
```

I think this is the same issue as #1133